user_guide/configuration.rst

```diff
@@ -245,41 +371,70 @@ You can also use ``composer.json`` to autoload, which will also allow for ``PSR-
     }
 
 If you add this to your ``composer.json`` file, then you won't need to specify autoloading in
-your ``behat.yml`` file:
+your ``behat.php`` file:
+
+.. code-block:: php
 
-.. code-block:: yaml
+    <?php
+    // behat.php
+    use Behat\Config\Config;
+    use Behat\Config\Profile;
+    use Behat\Config\Suite;
 
-    # behat.yml
+    $defaultProfile = new Profile('default')
+        ->withSuite(
+            new Suite('default')
+                ->withContexts('My\Application\Namespace\Bootstrap\FeatureContext')
+        )
+    ;
 
-    default:
-        suites:
-            default:
-                contexts: [My\Application\Namespace\Bootstrap\FeatureContext]
+    return new Config()
+        ->withProfile($defaultProfile)
+    ;
 
 Formatters
 ----------
 
 Default formatters can be enabled by specifying them in the profile.
 
-.. code-block:: yaml
+.. code-block:: php
 
-    # behat.yml
+    <?php
+    // behat.php
+    use Behat\Config\Config;
+    use Behat\Config\Profile;
+    use Behat\Config\Formatter\PrettyFormatter;
 
-    default:
-        formatters:
-            pretty: true
+    return new Config()
+        ->withProfile(
+            new Profile('default')
+                ->withFormatter(
+                    new PrettyFormatter()
+                )
+        )
+    ;
 
 Extensions
 ----------
 
 Extensions can be configured like this:
 
-.. code-block:: yaml
-
-    # behat.yml
-
-    default:
-        extensions:
-            Behat\MinkExtension:
-                base_url: http://www.example.com
-                selenium2: ~
+.. code-block:: php
+
+    <?php
+    // behat.php
+    use Behat\Config\Config;
+    use Behat\Config\Profile;
+    use Behat\Config\Formatter\PrettyFormatter;
+
+    return new Config()
+        ->withProfile(
+            new Profile('default')
+                >withExtension(
+                    new Extension('Behat\MinkExtension', [
+                        'base_url' => 'http://www.example.com',
+                        'selenium2' => '~',
```

Comment:
Actually, shouldn't this be `null` rather than the `'~`` string?

Reply:
it should. An unquoted `~` in Yaml is a `null` value.

Reply:
Yes, I think so too. I'm gonna change that.

Reply:
done